guppy/src/graph/build.rs

```diff
@@ -983,21 +981,16 @@ impl PackagePublishImpl {
     }
 }
 
-/// Replace backslashes in a relative path with forward slashes on Windows.
+/// Replace backslashes in a path with forward slashes on Windows.
 #[track_caller]
-fn convert_forward_slashes<'a>(rel_path: impl Into<Cow<'a, Utf8Path>>) -> Utf8PathBuf {
-    let rel_path = rel_path.into();
-    debug_assert!(
-        rel_path.is_relative(),
-        "path {} should be relative",
-        rel_path,
-    );
+fn convert_forward_slashes<'a>(path: impl Into<Cow<'a, Utf8Path>>) -> Utf8PathBuf {
```

Comment:
Ah, this is a really interesting consequence. I'm going to spend some time looking at the implications of this.

Reply:
Ah ok, so for absolute paths we don't convert to forward slashes on Windows: https://github.com/guppy-rs/guppy/blob/665f3f54acc1124c20b33103c713564530fadaf5/guppy/src/graph/build.rs#L546-L552

I would follow the same logic here.